java/src/main/java/ai/rapids/cudf/ParquetChunkedReader.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *  Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import java.io.File;
+
+/**
+ * TODO
+ */
+public class ParquetChunkedReader implements AutoCloseable {
+  static {
+    NativeDepsLoader.loadNativeDeps();
+  }
+
+
+
+  /**
+   * TODO
+   */
+  public ParquetChunkedReader(long chunkSizeByteLimit, File path) {
+    this(chunkSizeByteLimit, ParquetOptions.DEFAULT, path);
+  }
+
+  /**
+   * TODO
+   */
+  public ParquetChunkedReader(long chunkSizeByteLimit, ParquetOptions opts, File path) {
+    handle = create(chunkSizeByteLimit, opts.getIncludeColumnNames(), opts.getReadBinaryAsString(),
+        path.getAbsolutePath(), 0, 0, opts.timeUnit().typeId.getNativeId());
+  }
+
+  /**
+   * TODO
+   * @param chunkSizeByteLimit Byte limit (ex: 1MB=1048576)
+   * @param opts
+   * @param buffer
+   * @param offset
+   * @param len
+   */
+  public ParquetChunkedReader(long chunkSizeByteLimit, ParquetOptions opts, HostMemoryBuffer buffer,
+      long offset, long len) {
+    handle = create(chunkSizeByteLimit, opts.getIncludeColumnNames(), opts.getReadBinaryAsString(), null,
+        buffer.getAddress() + offset, len, opts.timeUnit().typeId.getNativeId());
+  }
+
+  /**
+   * TODO
```

Comment:
Please make sure to explain here on `readChunk` and in docs for this class that order that these APIs are supposed to be called in.  From https://github.com/NVIDIA/spark-rapids/pull/6934#discussion_r1013618442 it looks like it does not follow the standard way that java does this so we want to be 100% sure that we have documented it, or that we provide a way to make it match what java expects.